File: sdk-python/examples/quickstart.py
"""Runnable end-to-end example against a real hosted control-plane.

    BOXXKITE_BASE_URL=https://your-control-plane BOXXKITE_API_KEY=bxk_live_... \
        python examples/quickstart.py
"""

from __future__ import annotations

import os
import sys

from boxxkite_client import BoxxkiteApiError, BoxxkiteClient


def main() -> None:
    base_url = os.environ.get("BOXXKITE_BASE_URL")
    api_key = os.environ.get("BOXXKITE_API_KEY")
    if not base_url or not api_key:
        print("Set BOXXKITE_BASE_URL and BOXXKITE_API_KEY first.", file=sys.stderr)
        raise SystemExit(1)

    client = BoxxkiteClient(base_url=base_url, api_key=api_key)

    account = client.account()
    print(f"Signed in as {account['email']}")

    usage = client.usage()
    print(
        f"Usage: {usage['monthly_sandbox_hours_used']}/{usage['monthly_sandbox_hours_limit']} "
        f"sandbox-hours, {usage['concurrent_sandboxes']}/{usage['concurrent_sandboxes_limit']} concurrent"
    )

    try:
        with client.sandbox(label="sdk-quickstart") as sb:
            print(f"Created sandbox {sb.id}")

            result = sb.exec("python3 -c 'print(1 + 1)'")
            print(f"exec result: {result['stdout'].strip()}")

            sb.file_create("hello.txt", "hello from boxxkite-client\n")
            viewed = sb.view("hello.txt")
            print(f"file contents: {viewed['content'].strip()}")
        print("Sandbox destroyed.")
    except BoxxkiteApiError as exc:
        print(f"API error: {exc.message} [{exc.code}]", file=sys.stderr)
        raise SystemExit(1) from exc


if __name__ == "__main__":
    main()

File: sdk-js/examples/quickstart.ts
/**
 * Runnable end-to-end example against a real hosted control-plane.
 * Build first (relative imports resolve against dist/, not this .ts file):
 *
 *   npm run build
 *   BOXXKITE_BASE_URL=https://your-control-plane BOXXKITE_API_KEY=bxk_live_... \
 *       node dist/examples/quickstart.js
 */

import { BoxxkiteApiError, BoxxkiteClient } from "../src/index.js";

async function main() {
  const baseUrl = process.env.BOXXKITE_BASE_URL;
  const apiKey = process.env.BOXXKITE_API_KEY;
  if (!baseUrl || !apiKey) {
    console.error("Set BOXXKITE_BASE_URL and BOXXKITE_API_KEY first.");
    process.exit(1);
  }

  const client = new BoxxkiteClient({ baseUrl, apiKey });

  const account = await client.account();
  console.log(`Signed in as ${account.email}`);

  const usage = await client.usage();
  console.log(
    `Usage: ${usage.monthly_sandbox_hours_used}/${usage.monthly_sandbox_hours_limit} sandbox-hours, ` +
      `${usage.concurrent_sandboxes}/${usage.concurrent_sandboxes_limit} concurrent`,
  );

  try {
    await client.withSandbox(
      async (sb) => {
        console.log(`Created sandbox ${sb.id}`);

        const result = await sb.exec("python3 -c 'print(1 + 1)'");
        console.log(`exec result: ${result.stdout.trim()}`);

        await sb.fileCreate("hello.txt", "hello from boxxkite-client (js)\n");
        const viewed = await sb.view("hello.txt");
        console.log(`file contents: ${viewed.content.trim()}`);
      },
      { label: "sdk-js-quickstart" },
    );
    console.log("Sandbox destroyed.");
  } catch (err) {
    if (err instanceof BoxxkiteApiError) {
      console.error(`API error: ${err.message} [${err.code}]`);
      process.exit(1);
    }
    throw err;
  }
}

main();
